src/Test.UnitTests.Sarif/PropertiesDictionaryTests.cs

```diff
@@ -124,6 +130,69 @@ public void PropertiesDictionary_RoundTripStringToVersionMap()
             ((TypedPropertiesDictionary<Version>)properties[MapKey])[ValueKey].Should().Be(version);
         }
 
+        [Fact]
+        public void PropertiesDictionary_ConcurrentReadAndWrite()
+        {
+            var properties = new PropertiesDictionary();
+
+            Exception exception = Record.Exception(() =>
```

Comment:
>Exception [](http://example.com/codeflow?start=41&length=9)

Did you verify that this test actually fails when the fix isn't applied?

Reply:
yes this test can repro the error before change the type.